app/index.js

```diff
@@ -187,6 +189,46 @@ module.exports = yeoman.generators.Base.extend({
     );
   },
 
+  writeDependencies: function() {
+
+    this.options.plugins.push('grunt-karma');
+    if (this.options.coffee) {
+      this.options.plugins.push('coffee-script');
+    }
+
+    var done = this.async();
+    var packageJson = path.join(
+      this.options.cwd || process.cwd(),
+      'package.json'
+    );
+
+    var that = this;
+    fs.readFile(packageJson, { encoding: 'utf-8' }, function (err, content) {
+      var data;
+      if (err) {
+        that.log.error('Could not open package.json for reading.', err);
+        done();
+        return;
+      }
+
+      try {
+        data = JSON.parse(content);
+      } catch (err) {
+        that.log.error('Could not parse package.json.', err);
+        done();
+        return;
+      }
+
+      data.devDependencies = data.devDependencies || {};
+      Object.keys(that.options.plugins).forEach(function (t) {
```

Comment:
pretty plugins is an array. How about `this.options.plugins.forEach()`. Then you get the key directly and not an array index.